code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1336,3 +1336,837 @@
+/mob/living/carbon/human/proc/antag_hivemute()
+	set name = "Disrupt Hivenet User"
+	set desc = "Temporarily sever a Vaurca's connection to the Hivenet, preventing them from speaking. Failure will reveal your location to them!"
+	set category = "Hivenet"
+
+	var/list/available_vaurca
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, SPAN_WARNING("Your mind is dark, cut off from the unity of the Hive!"))
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You cannot attempt a Hivenet action so soon!"))
+		return
+	for(var/mob/living/carbon/human/player in living_mob_list)
+		if(player.stat == DEAD)
+			continue
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src)
+	LAZYADD(available_vaurca, "Cancel")
+	var/mob/living/carbon/human/target = input(src, "Select a target to disrupt", "Disrupt Hivenet User", null) as anything in available_vaurca
+	if(!istype(target))
+		return
+	var/obj/item/organ/internal/vaurca/neuralsocket/S = target.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(prob(70) || S.shielded)
+		to_chat(src, SPAN_DANGER("Your disruption attempt fails, [target]'s countermeasures repelling your assault! You can feel their defenses spring to life, tracing your location!"))
+		to_chat(target, SPAN_DANGER("You feel a presence attempting to block your neural socket, but your countermeasures repel it! Analysing the attack, you sense that it came from [get_area(src)]!"))
+	to_chat(target, SPAN_DANGER("Suddenly, you feel an attack on your systems, blocking your neural socket's ability to transmit to the Hivenet!"))
+	S.disrupted = 1
+	S.disrupttime = world.time + 10 MINUTES
+	host.last_action = world.time + 5 MINUTES
+
+/mob/living/carbon/human/proc/antag_hiveshock()
+	set name = "Hivenet Shock"
+	set desc = "Attempt to breach a Vaurca's defenses and deliver a painful neural shock. This has a 30% chance of success, and failure will reveal your location."
+	set category = "Hivenet"
+
+	var/list/available_vaurca
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!istype(host) || !(all_languages[LANGUAGE_VAURCA] in src.languages))
+		to_chat(src, SPAN_WARNING("Your mind is dark, unable to contact the Hive!"))
+		return
+	if(host.disrupted || host.muted)
+		to_chat(src, SPAN_WARNING("Your neural socket is unable to broadcast!"))
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before you can do another Hivenet action!"))
+		return
```

Comment:
A lot of these checks are copypasted all over these verbs, you should probably make them into a dynamic proc and just reuse that